pallets/issue/src/tests.rs

```diff
@@ -179,6 +198,33 @@ fn setup_execute(
 	issue_id
 }
 
+fn setup_execute_for_rate_limits(
+	issue_amount: Balance,
+	issue_fee: Balance,
+	griefing_collateral: Balance,
+	amount_transferred: Balance,
+) -> Result<H256, DispatchError> {
+	ext::vault_registry::get_active_vault_from_id::<Test>
+		.mock_safe(|_| MockResult::Return(Ok(init_zero_vault(VAULT))));
+	ext::vault_registry::issue_tokens::<Test>.mock_safe(|_, _| MockResult::Return(Ok(())));
+	ext::vault_registry::is_vault_liquidated::<Test>.mock_safe(|_| MockResult::Return(Ok(false)));
+
+	ext::fee::get_issue_fee::<Test>.mock_safe(move |_| MockResult::Return(Ok(wrapped(issue_fee))));
+	ext::fee::get_issue_griefing_collateral::<Test>
+		.mock_safe(move |_| MockResult::Return(Ok(griefing(griefing_collateral))));
+
+	let issue_id = request_issue_ok_with_address_for_rate_limits(USER, issue_amount, VAULT)?;
+	<security::Pallet<Test>>::set_active_block_number(5);
+
+	ext::stellar_relay::validate_stellar_transaction::<Test>
+		.mock_safe(move |_, _, _| MockResult::Return(Ok(())));
+	ext::currency::get_amount_from_transaction_envelope::<Test>.mock_safe(move |_, _, currency| {
+		MockResult::Return(Ok(Amount::new(amount_transferred, currency)))
+	});
+
+	Ok(issue_id)
+}
+
```

Comment:
Same here. Maybe I'm just blind. If we need a separate function please explain it to me haha.

Reply:
Ah wait, I think I get it now. You added a separate function to be able to get the result from it. 

But since it is basically just a duplicate I think I would prefer if we could have just one function for this. Maybe change it so that the functions you are interested in return a result and change their usages accordingly

Reply:
remove duplicated functions